### PR TITLE
dma_sam_xdmac - implemented dma device get_status()

### LINKED ISSUE #62003: SAM devices are missing the DMA API get_status() implementation
**Is your enhancement proposal related to a problem? Please describe.**
The DMA API allows to read the current status of a DMA channel. However this is not yet implemented for SAM XDMAC. 

**Describe the solution you'd like**
All required registers are defined. Would be a small fix to add this functionality.

**Describe alternatives you've considered**
I could patch the zephyr after downloading via git and use my patched version.

**Additional context**
I will create a fork and merge request with the required changes.

### PR DESCRIPTION
**Pull request by tswaehn**
added implementation for atmel sam xdmac - generic get_status() api 

- we can now read the current status of the dma through standard DMA API 

Fixes #62003

### CODE REVIEW

**Comment by tswaehn** [comment]
pipeline stopped with: 

`The self-hosted runner: zephyr-runner-linux-x64-4xlarge-deployment-g2tcl-q2sdx lost communication with the server`

is there something I can do?

**Comment by tswaehn** [comment]
tests seem to be complete.

**Comment by tswaehn** [comment]
modified commit message 
* author
* removed second line

as requested per code-guideline-check

File: drivers/dma/dma_sam_xdmac.c
Comment (pdgendt) [inline drivers/dma/dma_sam_xdmac.c:392]: Not sure why compliance checks aren't complaining, but some formatting needs to be done and some readability improvements:

```c
static int sam_xdmac_get_status(const struct device *dev, uint32_t channel,
				struct dma_status *status)
{
	const struct sam_xdmac_dev_cfg *const dev_cfg = dev->config;

	Xdmac * const xdmac = dev_cfg->regs;
	uint32_t chan_cfg = xdmac->XDMAC_CHID[channel].XDMAC_CC;
	uint32_t ublen = xdmac->XDMAC_CHID[channel].XDMAC_CUBC;

	if ((chan_cfg & XDMAC_CC_TYPE_Msk) == 0) {
		/* mem2mem */
		status->dir = MEMORY_TO_MEMORY;
	} else if ((chan_cfg & XDMAC_CC_DSYNC_Msk) == XDMAC_CC_DSYNC_MEM2PER) {
		/* mem2per */
		status->dir = MEMORY_TO_PERIPHERAL;
	} else {
		// per2mem
		status->dir = PERIPHERAL_TO_MEMORY;
	}

	status->busy = ((chan_cfg & XDMAC_CC_INITD_Msk) != 0) || (ublen > 0);
	status->pending_length = ublen;

	return 0;
}
```

**Review by pdgendt — CHANGES_REQUESTED** [review]

File: drivers/dma/dma_sam_xdmac.c
Comment (tswaehn) [reply drivers/dma/dma_sam_xdmac.c]: applied your changes

File: drivers/dma/dma_sam_xdmac.c
Comment (pdgendt) [inline drivers/dma/dma_sam_xdmac.c:379]: Nit: `status->busy` is a `bool`, you can drop the `? 1 : 0;`

Comment (tswaehn) [reply drivers/dma/dma_sam_xdmac.c]: done

**Review by pdgendt — DISMISSED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Comment by tswaehn** [comment]
@fabiobaltieri @pdgendt thank your for your approval. 

I am not authorized to merge - what is the typical procedure?

**Comment by fabiobaltieri** [comment]
> I am not authorized to merge - what is the typical procedure?

Someone from the release team will merge it for you, typically though we'd wait for assignee approval as well (that's @nandojve), keep in mind that many project members are on vacation these days so it may take another few days.

**Review by fabiobaltieri — CHANGES_REQUESTED** [review]
Actually wait a sec --- why is this trying to merge against v3.0-branch? v3.0 is unsupported, did you mean to file this against `main`?

**Comment by tswaehn** [comment]
we are using the v3.0.0 because we found main not stable regarding network stack and network drivers for the MICROCHIP SAM cpu. 

I will create a second pull request to fix this also in main, however we need this in v3.0.0, because that is the latest stable version, that we can use.

**Comment by fabiobaltieri** [comment]
No need to open a new PR, you can edit this one and change the base. We don't accept new PRs for unsupported releases though so if you want it on top of v3.0.0 you'll have to maintain a branch yourself.

**Comment by tswaehn** [comment]
So I guess it still would be ok to change the pull request to `branch-v2.7.5 (LTS)` - as this is officially the long term support branch?

**Comment by fabiobaltieri** [comment]
> I guess its still ok to change the pull request to `branch-v2.7.5 (LTS)` - as this is officially the long term support branch?

Yes but for that change this PR to merge against main and then we can use the backport tag so that the backport gets filed automatically as this gets merged on main.

**Comment by tswaehn** [comment]
ok understand. thank you for clarification. I will change to `main` then.

**Review by fabiobaltieri — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Review by nandojve — APPROVED** [review]
LGTM!

> we are using the v3.0.0 because we found main not stable regarding network stack and network drivers for the MICROCHIP SAM cO

Open an issue and decribe the details. I'm asking because I don't have any negative feedback about ethernet problems with SAM family. In fact, I've been using without any big issues since v2.7. What I noted is that sometimes people didn't explore all Kconfig options and sometimes buffers must to be increased.

**Comment by tswaehn** [comment]
We also noted that buffers needed to be increased. However regarding 24/7 stability we are not satisfied yet, maybe have to dig deeper. Thanks for your information.

**Pull request merged**
